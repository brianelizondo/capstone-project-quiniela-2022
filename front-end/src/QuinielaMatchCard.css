.QuinielaMatchCard-card{
    filter: drop-shadow(5px 5px 5px #000);
    margin-bottom: 50px;
}
.QuinielaMatchCard-match{
    color: gray;
    font-size: 15px;
    font-weight: 700;
    text-align: center;
    text-transform: uppercase;
}
.QuinielaMatchCard-teamlogo{
    text-align: center !important;
}
.QuinielaMatchCard-teamlogo img{
    aspect-ratio: auto;
    max-height: 100px;
    max-width: 90px;
}
.QuinielaMatchCard-teamnameA, .QuinielaMatchCard-teamnameB{
    font-size: 14px;
    text-transform: uppercase;
    color: black;
    padding-top: 30px;
}
.QuinielaMatchCard-teamnameA{
    text-align: left;
}
.QuinielaMatchCard-teamnameB{
    text-align: right;
}
.QuinielaMatchCard-result{
    font-family: 'Rubik', sans-serif;
    font-size: 50px;
    font-weight: 700;
    text-align: center;
    text-transform: uppercase;
}
.QuinielaMatchCard-points{
    font-family: 'Rubik', sans-serif;
    font-size: 26px;
    font-weight: 700;
    text-align: center;
    line-height: 26px;
}

.QuinielaMatchCard-points-text{
    color: gray;
    font-size: 10px;
    text-align: center;
    text-transform: uppercase;
    line-height: 12px;
}