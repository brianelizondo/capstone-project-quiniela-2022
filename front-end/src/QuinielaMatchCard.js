import React from 'react';
import { Card, Image, Row, Col } from 'react-bootstrap';
import './QuinielaMatchCard.css';

// card for the match in the quiniela
function MatchCard({ match, points }){
    const teamAName = match.teamA_name;
    const teamALogo = <Image src={ `/images/team_logo/${match.teamA_shortName}.png` } alt={ match.teamA_name } fluid={true} />;
    const teamBName = match.teamB_name;
    const teamBLogo = <Image src={ `/images/team_logo/${match.teamB_shortName}.png` } alt={ match.teamB_name } fluid={true} />;
    const matchResult = `${match.teamA_result} - ${match.teamB_result}`;
            
    return (
        <Card className="QuinielaMatchCard-card">
            <Card.Body>
                <Row>
                    <Col className="QuinielaMatchCard-match">Match { match.matchID }</Col>
                </Row>
                <Row>
                    <Col xs lg="2" className="QuinielaMatchCard-teamlogo">{ teamALogo }</Col>
                    <Col className="QuinielaMatchCard-teamnameA">{ teamAName }</Col>
                    <Col className="QuinielaMatchCard-result">{ matchResult }</Col>
                    <Col className="QuinielaMatchCard-teamnameB">{ teamBName }</Col>
                    <Col xs lg="2" className="QuinielaMatchCard-teamlogo">{ teamBLogo }</Col>
                </Row>
                <Row>
                    <Col className="QuinielaMatchCard-points">{ points }</Col>
                </Row>
                <Row>
                    <Col className="QuinielaMatchCard-points-text">points earned</Col>
                </Row>
            </Card.Body>
        </Card>
    );
}

export default MatchCard;