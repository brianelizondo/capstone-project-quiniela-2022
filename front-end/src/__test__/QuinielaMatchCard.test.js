import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { MemoryRouter } from "react-router-dom";

import QuinielaMatchCard from '../QuinielaMatchCard';

const match = {
    id: 1,
    matchID: 1,
    teamA_id: 1,
    teamA_name: "Qatar",
    teamA_shortName: "QAT",
    teamA_result: 0,
    teamB_id: 2,
    teamB_name: "Ecuador",
    teamB_shortName: "ECU",
    teamB_result: 0,
    points: 0
}

// Smoke Test
describe("Smoke Test", () => {
    it("renders 'QuinielaMatchCard' component without crashing", () => {
        render(
            <MemoryRouter>
                <QuinielaMatchCard match={match} />
            </MemoryRouter>
        );
        const textFound = screen.getByText("Match 1");
        expect(textFound).toBeInTheDocument();
    });
});

// Snapshot Tests
describe("Snapshot Test", () => {
    it("'QuinielaMatchCard' matches snapshot", () => {
        const {asFragment} = render(
            <MemoryRouter>
                <QuinielaMatchCard match={match} />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});