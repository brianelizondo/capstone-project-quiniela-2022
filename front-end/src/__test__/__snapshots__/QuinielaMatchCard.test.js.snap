// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 'QuinielaMatchCard' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="row"
      >
        <div
          class="col"
        >
          Match 1
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-lg-2 col"
        >
          <img
            alt="Qatar"
            class="img-fluid"
            src="/images/team_logo/QAT.png"
          />
        </div>
        <div
          class="col"
        >
          Qatar
        </div>
        <div
          class="col"
        >
          0 - 0
        </div>
        <div
          class="col"
        >
          Ecuador
        </div>
        <div
          class="col-lg-2 col"
        >
          <img
            alt="Ecuador"
            class="img-fluid"
            src="/images/team_logo/ECU.png"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
